spring-security/src/test/java/com/sap/cloud/security/spring/config/OAuth2ServiceConfigurationPropertiesTest.java

```diff
@@ -58,8 +61,38 @@ void setGetCertificateAndKey() {
 		assertEquals(ANY_VALUE, cutIas.getProperty(CERTIFICATE));
 	}
 
+	@Test
+	void setGetCertificateAndKeyXSUAA() {
+		cutXsuaa.setCertificate(ANY_VALUE);
+		cutXsuaa.setKey(ANY_VALUE);
+		cutXsuaa.setClientId(ANY_VALUE);
+		cutXsuaa.setClientSecret(ANY_VALUE); // to make sure that getClientIdentity uses ClientCertificate impl as
+												// default when possible
+		assertEquals(ANY_VALUE, cutXsuaa.getClientIdentity().getCertificate());
+		assertEquals(ANY_VALUE, cutXsuaa.getClientIdentity().getKey());
+		assertTrue(cutXsuaa.hasProperty(CERTIFICATE));
+		assertEquals(ANY_VALUE, cutXsuaa.getProperty(CERTIFICATE));
+		assertTrue(cutXsuaa.hasProperty(KEY));
+		assertEquals(ANY_VALUE, cutXsuaa.getProperty(KEY));
+		assertTrue(cutXsuaa.getClientIdentity().isCertificateBased());
+	}
+
+	@Test
+	void getClientIdentityResolvesToClientCredentials() {
+		cutIas.setClientId(ANY_VALUE);
+		cutIas.setClientSecret(ANY_VALUE);
+		assertFalse(cutIas.getClientIdentity().isCertificateBased());
+
+		cutXsuaa.setClientId(ANY_VALUE);
```

Comment:
whats the difference between cutXsuaa and cutIas?

Reply:
good point, the only difference is the service field

Reply:
leaving it in as discussed